### PR TITLE
Fix different engagement emails to the same user not sent when trigge…

### LINKED ISSUE #1329: Engagement emails: dupcheck prevents the sending of further different emails to the same user with the same trigger
### Reproduction Steps

+ Create an engagement email A
+ Create an engagement email B
+ Create a course C with only one lesson
+ Create an engagement E1 and set it up to send the email A to the user U who enrolls C
+ Create an engagement E2 and set it up to send the email B to the user U who completes C
+ Enroll an user into C
+ Check the email A has been sent (check the engagement-emails log as well)
+ As the user U complete C
+ Check whether the email B has been sent (check the engagement-emails log as well)

### Expected Behavior

+ I'd expect the email B to be sent

### Actual Behavior

+ The email B has not been sent because of the dupcheck - although the message logged is wrong because of another bug.

### Error Messages / Logs
engagement-emails log

```
09-16-2020 @ 16:14:22 - Email #20 to user #4937 triggered by #4821 not sent not sent due to user enrollment issues.
```

Please also note that in the message above there are 2 errors:
1) 20 is the user id, and 4937 is the email id, they're reversed
2) The real reason of the email not sent is the dupcheck not the enrollment issue.

### System and Environment Information

```
LifterLMS 4.4.2
```

</details>

This issue has be recreated:
+ [x] Locally
+ [x] On a staging site
+ [x] On a production website
+ [x] With only LifterLMS and a default theme

### Browser, Device, and Operating System Information

n/a

### Related User Information
HS-134997

p.s.
I must say I did a bad job in reviewing #1310 with the aggravating circumstance that nobody requested my review. Sorry

### PR DESCRIPTION
**Pull request by eri-trabiccolo**
…red by the same post

## Description

Fixes #1329 

## How has this been tested?
Old and new unit tests

## Screenshots 
n/a 

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: includes/class.llms.engagements.php [hunk 1]
```diff
@@ -18,6 +18,9 @@
  * @since 2.3.0
  * @since 3.30.3 Fixed spelling errors.
  * @since 3.39.0 Added `llms_rest_student_registered` as action hook.
+ * @since 4.4.1 Avoided emails to be sent to students whose enrollment is not active in the related course or membership which triggered the engagement.
+ *              Also engagement email related logs logged to a separate logfile, engagement-emails in favor of the main llms log.
+ * @since [version] Fixed different emails triggered by the same related post (trigger) not sent because of a wrong duplicate check.
```
Comment (thomasplevy) [inline includes/class.llms.engagements.php]:
```suggestion
```

Unnecssary duplicates

**Review by thomasplevy — CHANGES_REQUESTED** [review]
minor doc, looks good otherwise

**Comment by thomasplevy** [comment]
@eri-trabiccolo I accidentally merged into trunk (didn't realize it wasn't dev) so reverted but I cannot reopen the PR and change the base. Can you please resubmit to dev.

Thanks

**Comment by thomasplevy** [comment]
Nevermind resubmitted myself (I didn't know I could do that, or rather, it never occurred to me that I could)

**Comment by eri-trabiccolo** [comment]
Didn't know that either 🤣

**Comment by thomasplevy** [comment]
Yea, you can open a PR against any existing (public) fork. Pretty cool

**Comment by eri-trabiccolo** [comment]
Mmm but it's something recent I guess, as when we switched to the trunk branch we had to ask contributors to submit their PRs again...
Or maybe we just didn't notice it was already possibile. :)

**Comment by thomasplevy** [comment]
Bingo on your or (probably)

**Pull request merged**